File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans;
        unordered_map <int, int> hash;
        int n= nums.size();
        for (int i=0; i<n; i++){
            hash[nums[i]]++;
        }

        for (auto it: hash){
            if (it.second > (n/2)){
                ans= it.first;
                break;
            }
        }

        return ans;

    }
};

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        c=0
        e=0
        n= len(nums)

        for i in range (n):
            if (c==0):
                c=1
                e=nums[i]

            else:
                if (nums[i]== e):
                    c+=1
                else:
                    c-=1

        return e

                